benchexec/tablegenerator/react-table/src/components/TaskDefinitionViewer.js

```diff
@@ -0,0 +1,114 @@
+/* SPDX-License-Identifier: Apache-2.0
+ *
+ * BenchExec is a framework for reliable benchmarking.
+ * This file is part of BenchExec.
+ * Copyright (C) Dirk Beyer. All rights reserved.
+ */
+import React from "react";
+import yamlParser from "yaml";
+
+/** Special view for YAML files in the LinkOverlay component. */
+export default class TaskDefinitionViewer extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      splitterTag: "<splitter#9d81y23>",
+      fileTag: "<file#092nt43>",
+      content: this.props.yamlText,
+    };
+  }
+
+  componentDidMount() {
+    this.prepareTextForRendering();
+  }
+
+  componentDidUpdate(prevProps) {
+    if (prevProps.yamlText !== this.props.yamlText) {
+      this.prepareTextForRendering();
+    }
+  }
+
+  /**
+   * Parses the YAML file and encloses all input and property files with a fileTag as well as a splitterTag,
+   * so they can be rendered separately as links. Takes the following assumptions:
+   * input_files is either a string or a list of strings
+   * properties is a list of dicts, each with a "property_file" key
+   */
+  prepareTextForRendering = () => {
+    const yamlObj = yamlParser.parseDocument(this.props.yamlText);
+
+    const inputFiles = yamlObj.get("input_files");
+    if (inputFiles) {
+      if (Array.isArray(inputFiles.items)) {
+        inputFiles.items.forEach(inputFileItem => {
+          inputFileItem.value = this.encloseFileInTags(inputFileItem.value);
+        });
+      } else {
+        yamlObj.set("input_files", this.encloseFileInTags(inputFiles));
+      }
+    }
+
+    const properties = yamlObj.get("properties");
+    if (properties) {
+      if (Array.isArray(properties.items)) {
+        properties.items.forEach(property => {
+          if (Array.isArray(property.items)) {
+            property.items.forEach(propertyItem => {
+              if (propertyItem.key.value === "property_file") {
+                propertyItem.value.value = this.encloseFileInTags(
+                  propertyItem.value.value,
+                );
+              }
+            });
+          }
+        });
+      }
+    }
+
+    this.setState({ content: yamlObj.toString() });
+  };
+
+  encloseFileInTags = fileName => {
+    return (
+      this.state.splitterTag +
+      this.state.fileTag +
+      fileName +
+      this.state.fileTag +
+      this.state.splitterTag
+    );
+  };
+
+  render() {
+    const contentBySplitter = this.state.content.split(this.state.splitterTag);
```

Comment:
I don't see the benefit of modifying the `yamlObj` in `prepareTextForRendering` and wrapping the filename in tags only to split them again here. 
Is there an argument against building an array of objects in `prepareTextForRendering` instead of modifying the mutable properties and then calling `toString` ?
I think we could build objects of this structure 
```
[
   {
      inputFiles: [...],
      propertyFiles: [...],
   },...
]
```
and then iterate through both arrays in here. 
Or is there something that i am missing?

Reply:
The reason I implemented it like this was simplicity. It should be possible to build an array like you suggested, however the library that is used stores the yaml file in a pretty complex object. To keep the formation (like spaces or empty lines, brackets vs. new lines, multiline strings etc.) of the yaml file, there will be a lot of code necessary which the `toString()` method automatically does for us here.

Also the current code works with yaml files that aren't of the standard for the task definition files by just ignoring everything except for the important parts. If we rebuild the yaml file manually, we will have to take all possibilities into consideration to also rebuild the (in this case) not important parts correctly.

Reply:
I see. That makes sense. 
One more question: Is the splitter tag needed? Wouldn't it work with the file tag alone?

Reply:
Hm, the reason why I used both tags is that the splitter tag is removed from the string by the `split()` method. The file tag is then used to identify the elements of the array that are supposed to be linked.
It would be possible to use only one tag and leave them in the array by using regex with the `split()` method, but this would result in an array like this:
`[..., "input_files: ", "<fileTag>", "inputFileName", "<fileTag>", ...]`
So then additional code will be necessary to only link the strings with preceding and following `<fileTag>` elements in the array and not to render the `<fileTag>` strings. I don't think this will have much benefit for the code quality.